pkg/occlient/occlient.go

```diff
@@ -2100,19 +2097,19 @@ func (c *Client) updateDCOfComponent(componentName, applicationName string, dcUp
 		return errors.Wrapf(err, "Unable to locate DeploymentConfig for component %s of application %s", componentName, applicationName)
 	}
 
-	if dcUpdater != nil {
-		err = dcUpdater(dc)
+	if dcPatchProvider != nil {
+		patch, err := dcPatchProvider(dc)
 		if err != nil {
-			return errors.Wrap(err, "Unable to update the DeploymentConfig")
+			return errors.Wrap(err, "Unable to create a patch for the DeploymentConfig")
 		}
-	} else {
-		return errors.Wrapf(err, "dcUpdater was not properly set - this is definitely an implementation issue")
-	}
 
-	// update the DeploymentConfig with the secret
-	_, err = c.appsClient.DeploymentConfigs(c.Namespace).Update(dc)
-	if err != nil {
-		return errors.Wrapf(err, "DeploymentConfig not updated %s", dc.Name)
+		// update the DeploymentConfig with the secret
+		_, err = c.appsClient.DeploymentConfigs(c.Namespace).Patch(dcName, types.JSONPatchType, []byte(patch))
+		if err != nil {
+			return errors.Wrapf(err, "DeploymentConfig not updated %s", dc.Name)
+		}
+	} else {
+		return errors.Wrapf(err, "dcPatch was not properly set - this is definitely an implementation issue")
```

Comment:
error message seems kind of odd, remove `definitely an implemention issue` maybe?

Reply:
Fixed